inst/EpiModel 2.0 Introduction.rmd

```diff
@@ -0,0 +1,254 @@
+
+```{r echo = TRUE, eval = FALSE}
+?tergmLite::tergmLite-package
+```
+
+The use of ```tergmLite``` is fairly straightforward: the input parameter ```tergmLite``` is passed into a ```control.net``` object as a logical indicating whether to use the full network object (```tergmLite == FALSE```) or a pared down version of the network (```tergmLite == TRUE```). As an example:
+
+```{r echo = TRUE, eval = FALSE}
+nw <- network_initialize(100)
+formation <- ~edges
+target.stats <- 30
+coef.diss <- dissolution_coefs(dissolution = ~ offset(edges), duration = 25)
+est <- netest(nw, formation, target.stats, coef.diss)
+
+init <- init.net(i.num = 5)
+param <- param.net(inf.prob = 0.1)
+
+## tergmLite = FALSE: use full network object
+control <- control.net(type = "SI", nsteps = 100, nsims = 1, tergmLite = FALSE)
+sim1 <- netsim(est, param, init, control)
+
+## tergmLite = TRUE: use reduced cross-sectional edgelist
+control <- control.net(type = "SI", nsteps = 100, nsims = 1, tergmLite = TRUE)
+sim2 <- netsim(est, param, init, control)
+```
+
+## Network Update
+
+Prior to EpiModel 2.0, the base functions of EpiModel were built using functions both from ```EpiModel``` and ```network``` packages. This intermingling of functions made it difficult for the end user to tease out where network functionality ended and EpiModel began, hampering understanding of the EpiModel workflow and the ability to build module extensions. In order to make the distinction between the two packages explicit and to make the purely EpiModel workflow clearer, a new base function, ```nwupdate.net``` has been introduced. This function pulls all interactions with the network object from base functions into ```nwupdate.net``` and is run at the end of the simulation. By segmenting the EpiModel workflow in this way, base EpiModel modules are now easier to navigate and make clear the output needed for module extensions.
+
+## Helper Functions
+
+Several helper functions have been added to EpiModel that make it clear the ```$``` indexing operations that are being done on the master ```dat``` object during simulation. These functions belong to the ```get``` and ```set``` family of functions and act on the various lists contained within the master ```dat``` object, including but not limited to the attribute, epi, control and param lists. These functions take as input, at mininum, the master ```dat``` object and the item (as a character string) to "get" or "set" ("age" or "group" in the attribute list or "inf.prob" in the parameter list for exampe).
+
+### Set Functions
+
+The ```set``` group of functions sets the value of an item on a sublist of the ```dat``` object; these functions are often used at the end of a modules sequence to update attributes or for tracking of epidemic statistics. As an example: 
+
+```{r echo = TRUE, eval = FALSE}
+## Code snippet from one-Group infection module:
+dat <- set_epi(dat, "si.flow", at, nInf)
+```
```

Comment:
shouldn't we add the corresponding EpiModel 1.8 code as an comparison?

`dat[["epi"]][["si.flow"]][at] <- nInf`

Maybe with a comment explaining that it should not be done in EpiModel 2.0
I am actually not sure if it would clarify the changes or confuses users.

Reply:
@AdrienLeGuillou updated with latest commit but moved to code migration section. Let me know if this was what you were thinking of.